_posts/2017-09-05-Browser Security Features.md

```diff
@@ -0,0 +1,290 @@
+ * `unsafe-inline` lets you use inline javascript and CSS (although it's preferable to use a nonce)
+ * `unsafe-eval` allows the use of `eval()`, `setTimeout(String)`, `setInterval(String)` and `new Function(String)`.
+ There's a reason it has "unsafe" in its name though: these functions are typically used as attack vectors for XSS.
+ * `https:` allow content loaded from anywhere, as long as it's served over https
+ * `example.com` allow content loaded from anywhere on example.com, both http and https
+ * You can also use wildcards to control which origins are allowed. 
+ E.g. `*://*.example.com:*` will allow resources to be loaded from all *subdomains* of example.com, using any scheme and port.
+ Note that it won't allow you to load resources from example.com itself.
+ * `nonce-...` Allows you to specify a [nonce](#nonce). Scripts or styles that have this nonce are then allowed to execute.
+It's also important to note that you can pass multiple values to these directives: `self https://example.com` will allow resources to be loaded both from the domain itself as well as from https://example.com.
+
+You can use these to define `default-src`, but CSP gives you more fine-grained control over where each type of resource can be loaded from.
+For that you need to use the following properties instead:
+ * `script-src` - Javascript.
+ * `style-src` - CSS
+ * `img-src` - images
+ * `font-src` - fonts
+ * `object-src` - objects (e.g. `<object>`, `<embed>`, ...)
+ * `object-src` - media such as `<audio>` and `<video>` elements
+ * `connect-src` - where the page can connect to using `XmlHttpRequest`, `WebSocket` or `EventSource`.
+ The browser will immediately return a 400 status code when your page attempts to connect to a non-valid domain.
+ * `frame-src` - Specify which locations can be embedded in a `<frame>` and `<iframe>`
+ * `worker-src` - Worker scripts
+ * `child-src` - Is either deprecated or serves as fallback for `frame-src` and `worker-src`, depending on the browser and the CSP level implemented.
+ 
+### Navigation directives
+These directives tell the browser what kind of navigation is allowed:
+* `form-action` limits to where forms can be submitted
+* `frame-ancestors` specifies who may embed pages using elements such as `<frame>`, `<iframe>`, `<embed>`, `<object>` and `<applet>` 
+
+### Other directives
+Besides these, CSP allows for quite a few other directives:
+ * `report-uri` works the same way as the `report-uri` directive of HPKP
+ * `require-sri-for` allows you to force the use or Subresource Integrity ((SRI)[#Subresource-integrity]) for stylesheets, scripts or both. 
+ Allowed values are `script` and `style` (or both).
+ * `base-uri` defines which URLs you can use in the `<base>` element
+ * `sandbox` to enable a sandbox for requested resources 
+ (have a look at [Mozilla's documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/sandbox){: target='_blank' } for more information)
+ 
+### Nonce
```

Comment:
Maybe add that it is pronounced "/nɒns/" (nance) and stands for "number (used) once".